### PR TITLE
Added support and demo for using PFQueryCollectionViewController from Storyboards and XIBs.

### LINKED ISSUE #38: PFQueryCollectionViewController not fetching PFObjects in Swift
I have implemented it and initialized it this way 
    required init(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        
        self.parseClassName = "userPhoto";
        self.pullToRefreshEnabled = true
        self.paginationEnabled = true
        self.loadingViewEnabled = true
    }

objectsDidLoad method is reporting nil as error and objects as nil too. Also i have tried to set the view controller's user defined run time attribute of parseUserClass = "userPhoto" but it hasn't helped. Nothing is showing up or being fetched.

### PR DESCRIPTION
**Pull request by nlutsenko-fb**
Fixes #38 

cc @grantland @hallucinogen @stanley-parse

### CODE REVIEW

File: ParseUIDemo/Classes/CustomViewControllers/QueryCollectionViewController/StoryboardCollectionViewController.m [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    const CGRect bounds = UIEdgeInsetsInsetRect(self.view.bounds, layout.sectionInset);
+    CGFloat sideSize = MIN(CGRectGetWidth(bounds), CGRectGetHeight(bounds)) / 2.0f - layout.minimumInteritemSpacing;
+    layout.itemSize = CGSizeMake(sideSize, sideSize);
+}
+
+
+#pragma mark -
+#pragma mark Query
+
+- (void)_setupCollectionQuery {
```
Comment (hallucinogen) [inline ParseUIDemo/Classes/CustomViewControllers/QueryCollectionViewController/StoryboardCollectionViewController.m]: I'll keep it in `initWithCoder` for consistencies with other examples.

**Comment by hallucinogen** [comment]
Otherwise, looks good to me! Ship!

**Comment by jccf091** [comment]
I am very new to iOS. After a lot of googling, i still can't find complete example of how to use ParseUI. Would your guys write a demo in swift. 
It will be appreciated.

**Comment by jccf091** [comment]
```swift
class LatestPostCollectionViewController: PFQueryCollectionViewController {
  
  required init(coder aDecoder: NSCoder) {
    super.init(coder:aDecoder)
    
    self.setupCollectionQuery()
  }
  
  override func viewDidLoad() {
    super.viewDidLoad()
    
    var layout: UICollectionViewFlowLayout = self.collectionViewLayout as UICollectionViewFlowLayout
    layout.sectionInset = UIEdgeInsetsMake(0.0, 10.0, 0.0, 10.0)
    layout.minimumInteritemSpacing = 5.0
  }
  
  override func viewWillLayoutSubviews() {
    super.viewWillLayoutSubviews()
    
    var layout: UICollectionViewFlowLayout = self.collectionViewLayout as UICollectionViewFlowLayout
    
    let bounds = UIEdgeInsetsInsetRect(self.view.bounds, layout.sectionInset)
    var sideSize: CGFloat  = (min(CGRectGetWidth(bounds), CGRectGetHeight(bounds))/2.0 - layout.minimumInteritemSpacing)
    
    
    layout.itemSize = CGSizeMake(sideSize, sideSize)
  }
 
  func setupCollectionQuery() {
    self.parseClassName = "Post"
    self.pullToRefreshEnabled = true
    self.paginationEnabled = true
  }
  
  override func queryForCollection() -> PFQuery! {
    var query = super.queryForCollection()
    query.orderByAscending("likes_count")
    return query
  }
  
  override func collectionView(collectionView: UICollectionView, cellForItemAtIndexPath indexPath: NSIndexPath, object: PFObject) -> PFCollectionViewCell {
    let post = object as Post
    
    let cell =  super.collectionView(collectionView, cellForItemAtIndexPath: indexPath, object: object)
    cell.contentView.layer.borderWidth = 1.0
    cell.imageView.imageURL = NSURL(string: post.lowResolutionImageURL)
  
    return cell
  }  
}
```
 cell.imageView.imageURL  is a stuff from AsyncImageView which extends UIImageView with imageURL
ios simulator screen shot 30 jan 2015 4 06 11 pm

If I make imageView inside PFCollectionViewCell adapt contentMode to size of the image( which is your code in pull request #40 ), all the image will not be displayed.

**Comment by sheharyar-rr** [comment]
cell.clipToBounds = True will help, although i wasn't able to successfully resize the pictures as you said, they simply stop displaying at all. And there are repeating pictures

**Comment by emrecaliskan** [comment]
Ok, so here is what I did in swift to make things work for me. As I mentioned in the previous comment, I don't have much experience formatting programatically and wanted to use the interface builder to format the PFQuery collection. Anyway, here are the steps I took:

1 Added the following to the beginning "PFQueryCollectionViewController" implementation file. 

```swift
- (instancetype)initWithCoder:(NSCoder *)decoder {
    // initWithCoder is usually a parallel designated initializer, as is the case here
    // It's used by storyboard
    if (self = [super initWithCoder:decoder]) {
        [self _setupWithClassName:nil];
    }
    return self;
}

- (instancetype)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil {
    // This is used by interface builder
    if (self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil]) {
        [self _setupWithClassName:nil];
    }
    return self;
}
```


2 Created a collection controller on the story board. Created my custom collection view controller class (subclass of PFQueryCollectionViewController) as show below and linked the class to the collection controller I just created. After step 1, Xcode initializes everything using the storyboard and interface builder, so the custom controller class becomes very very simple.

```swift
import UIKit

class CustomPFCollectionViewController: PFQueryCollectionViewController {
    
    let reusableCellID = "collectionViewCell"
    
    required init(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        self.parseClassName = "Item"
    }
    
    
    override func collectionView(collectionView: UICollectionView!, cellForItemAtIndexPath indexPath: NSIndexPath!, object: PFObject!) -> PFCollectionViewCell! {
        
        
        let cell = collectionView.dequeueReusableCellWithReuseIdentifier(reusableCellID, forIndexPath: indexPath) as? CustomPFCollectionViewCell
        
        cell?.titleLabel.text = object["title"] as NSString
        cell?.myImageView.file = (object["images"] as [PFFile]).first
        cell?.myImageView.loadInBackground()

        
        return cell
    }
}

```

3 Since step 2 uses a custom cell class, I created that class as shown below. Then linked the class to the cell through the interface builder.

```swift
import UIKit

class CustomPFCollectionViewCell: PFCollectionViewCell {

    @IBOutlet var myImageView: PFImageView!
    @IBOutlet var titleLabel: UILabel!
}


```


I believe the most important thing was step 1, where the storyboard and interface builder didn't have initializers to use from the PFQueryCollectionViewController class. I may be wrong, but everything works so I'll stick with it for now.

**Comment by jccf091** [comment]
@emrecaliskan Thanks!!!!! It works. You just save me!

**Comment by sheharyar-rr** [comment]
@emrecaliskan @jccf091  Sorry for the noobish question but how did you change the implementation file in the ParseUI framework? Can you explain step by step?

**Comment by jccf091** [comment]
@sheharyar-rr 
Ya. You need to change the PFQueryCollectionViewController.m file just like @emrecaliskan said in step one.
you should use ParseUI which is downloaded from Github. So that you can change the file.

**Comment by emrecaliskan** [comment]
@sheharyar-rr 

I'm using Cocoapods, so I simply see and edit the file as shown below.

screen shot 2015-01-30 at 11 10 39 am

**Comment by nlutsenko-fb** [comment]
Yep - the most important changes are in this PR - this brings those 2 methods (that enable Storyboard and XIBs (interface builder)) to PFQueryCollectionViewController.

Thanks guys, I am merging this upstream in a bit.

**Comment by nlutsenko-fb** [comment]
@emrecaliskan 
Your Swift example looks great!
We probably should add that to a list of examples in ParseUIDemo to help others to get started.
Looking forward to see a Pull Request with that Demo from you :)

**Pull request merged**